src/metabase/models/action.clj

```diff
@@ -282,7 +289,20 @@
                                 (map #(dissoc % ::pk? ::field-id)))]
           (cond-> (assoc action :database_id (model-id->db-id (:model_id action)))
             (seq implicit-params)
-            (assoc :parameters implicit-params)))
+            (-> (assoc :parameters implicit-params)
+                (update-in [:visualization_settings :fields]
+                           (fn [fields]
+                             (let [param-ids (map :id implicit-params)]
+                               (let [fields (->> (or fields {})
+                                                 ;; remove entries that don't match params
+                                                 (m/filter-keys (set param-ids)))]
```

Comment:
in which case this filter is needed?

Reply:
1. User creates an model for a table with columns A and B, and creates basic actions. Then edits visualization_settings.fields for A.
2. The table changes to only have column B, with no column A anymore. 

`param-ids` will now only include B, and `fields` will continue to include A and B. A needs to be filtered out of `fields`.